plugins/providers/shield/config.go

```diff
@@ -0,0 +1,132 @@
+package shield
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/mitchellh/mapstructure"
+	"github.com/odpf/guardian/domain"
+)
+
+const (
+	RoleMember = "users"
+	RoleAdmin  = "admins"
+
+	AccountTypeUser = "user"
+)
+
+type Credentials struct {
+	Host      string `json:"host" mapstructure:"host" validate:"required"`
+	AuthEmail string `json:"auth_email" mapstructure:"auth_email" validate:"required"`
```

Comment:
can you explain a little on how we authenticate to shield?

Reply:
We'll be creating shield user have Org admin access specifically for Guardian calls and that user being used in Policy credentials.